### PR TITLE
Fix \vec

### PR DESCRIPTION
**Pull request by ronkok**
For the accent `\vec`, replace the combining font glyph with an SVG.

### CODE REVIEW

**Comment by ronkok** [comment]
This PR is written to fix issues #735 and #996.

`\vec` uses a combining character and does not have a character preceding it within its span. Safari, unlike other browsers, does not recognize the negative `xMin` value for such a character, so it renders `\vec` too far to the right.

This PR replaces the glyph with an SVG whose path is copied from the glyph. Let’s all just agree that this is an ugly hack. It is defensible only because user agents cannot be trusted to handle combining characters consistently.

The people over at MathJax must have come to the same conclusion. [Their approach](https://github.com/mathjax/MathJax/pull/1796) is also, by their own admission, a hack.

**Comment by kevinbarabash** [comment]
I missed `HorizontalBraces`... PR in coming.

**Comment by kevinbarabash** [comment]
accents

KaTeX is in red, LaTeX is in green.  The accent spacing looks really consistent.

**Comment by ronkok** [comment]
There was an alternative approach that I examined. If one prepends a non-breaking space before the combining character U+20D7, Safari will then treat it in the same way as the other browsers that I tested.

The problem then shifts to the vertical alignment. When the arrow is so treated, none of the browsers place in the vertical position consistent with the glyphs `yMin` value. It's lower. By my eyeball test, it looks to be vertically centered on the x-height. But that is not documented anywhere that I can find. Rather than depend on something so dicey, I wrote the SVG.

**Comment by kevinbarabash** [comment]
> Rather than depend on something so dicey, I wrote the SVG.

Good call.  I should be able to review this tomorrow.

**Comment by edemaine** [comment]
@ronkok Indeed, I'd tried that hack (based on the MathJax hack, which I think you originally pointed me to) in an intermediate version of #802, too, and also couldn't get it to work.  (I think MathJax has the luxury of doing different things depending on the browser. 
 KaTeX can't do that, offline...)  Thanks for finding another way!  Given that most (long) accents are now drawn in SVG, this seems like a nice fix.

Just to point out: There is an alternative, to modify the font so that it has a non-combining version of the character.  This is why `\acute` and all other accents work (I assume) in Safari.  But I have never touched the fonts, so don't know how hard this approach is, or which is preferable.

**Comment by ronkok** [comment]
> There is an alternative, to modify the font so that it has a non-combining version of the character.

That would be better.  But fonts are beyond me. Hopefully, someone can make the font modification in some future PR. This PR can be a good enough temporary measure.

File: src/buildCommon.js [hunk 1]
```diff
@@ -621,6 +621,30 @@ const fontMap: {[string]: {| variant: string, fontName: string |}} = {
     },
 };
 
+const svgData: {
+    [string]: ([string, number, number])
```
[the remark this answers is not in the document]
Comment (marcianx) [reply src/buildCommon.js]: I don't think flow will infer a tuple type, but a more type-unsafe heterogeneous array. One could try to check.

My personal bias is to favor explicit typing to make the intention clear and better documented.

[the remark this answers is not in the document]
Comment (ronkok) [reply src/buildCommon.js]: I was following the example set by @marcianx during the conversion of `stretchy.js` to flow.

[the remark this answers is not in the document]
Comment (kevinbarabash) [reply src/buildCommon.js]: We can totally leave it in.  It gets compiled away anyways.

**Comment by kevinbarabash** [comment]
I restarted the build.

**Pull request merged**